.github/workflows/backend-tests.yml

```diff
@@ -6,15 +6,67 @@ on:
 
   workflow_dispatch:
 
+env:
+  BACKEND_DEBUG: 0
+  POSTGRES_PASSWORD: example
+  POSTGRES_USER: example
+  DB_HOST: db
+  DB_NAME: postgres
+  FRONTEND_TARGET: production
+  NODE_IMAGE: lts-alpine
+  JWT_SECRET_KEY: example
+  DISABLE_AUTHENTICATION: true
+
+
 jobs:
   build:
     runs-on: ubuntu-latest
+    timeout-minutes: 20
     steps:
       - uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+      - uses: actions/setup-node@v3
+        with:
+          node-version: '14'
+          cache: 'npm'
+          cache-dependency-path: '**/package-lock.json'
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v1
+      - name: Cache Docker layers
+        uses: actions/cache@v2
+        with:
+          path: /tmp/.buildx-cache
+          key: docker-images
+          restore-keys: |
+            docker-images
```

Comment:
```suggestion
```
From my understanding, `restore-keys` is only useful if there is no match on `docker-images` see https://docs.github.com/en/actions/using-workflows/caching-dependencies-to-speed-up-workflows